src/planner/parabolic/ParabolicTimer.cpp

```diff
@@ -113,6 +115,43 @@ std::unique_ptr<aikido::trajectory::Spline> computeParabolicTiming(
   return outputTrajectory;
 }
 
+//==============================================================================
+std::unique_ptr<aikido::trajectory::Spline> computeParabolicTiming(
+    const aikido::trajectory::Spline& _inputTrajectory,
+    const Eigen::VectorXd& _maxVelocity,
+    const Eigen::VectorXd& _maxAcceleration)
+{
+  const auto stateSpace = _inputTrajectory.getStateSpace();
+  const auto dimension = stateSpace->getDimension();
+
+  if (static_cast<std::size_t>(_maxVelocity.size()) != dimension)
+    throw std::invalid_argument("Velocity limits have wrong dimension.");
+
+  if (static_cast<std::size_t>(_maxAcceleration.size()) != dimension)
+    throw std::invalid_argument("Acceleration limits have wrong dimension.");
+
+  for (std::size_t i = 0; i < dimension; ++i)
+  {
+    if (_maxVelocity[i] <= 0.)
+      throw std::invalid_argument("Velocity limits must be positive.");
+    if (!std::isfinite(_maxVelocity[i]))
+      throw std::invalid_argument("Velocity limits must be finite.");
+
+    if (_maxAcceleration[i] <= 0.)
+      throw std::invalid_argument("Acceleration limits must be positive.");
+    if (!std::isfinite(_maxAcceleration[i]))
+      throw std::invalid_argument("Acceleration limits must be finite.");
+  }
+
+  double startTime = _inputTrajectory.getStartTime();
+  auto dynamicPath = detail::convertToDynamicPath(
+      _inputTrajectory, _maxVelocity, _maxAcceleration, true);
+
+  auto outputTrajectory
+      = detail::convertToSpline(*dynamicPath, startTime, stateSpace);
```

Comment:
Is `dynamicPath` always not `nullptr`? If so, please throw an exception if `dynamicPath` is `nullptr`. 

Otherwise, return `nullptr` as output trajectory if `dynamicPath` is `nullptr`.

Reply:
The current implementation of `convertToDynamicPath` guarantees that `dynamicPath` will never be `nullptr`.
Do you think we shall add the check in case any potential change will happen in the future?